### ISSUE TITLE
Lumines Electronic Symphony [PCSE00009]

### ISSUE THREAD

**Issue by othmanead** [Vita3K/compatibility#693]
# App summary
- App name: Lumines Electronic Symphony
- App serial: PCSE00009
- App version: 1.0

# Vita3K summary
- Version: v0.1.5
- Build number: 2647
- Commit hash: https://github.com/vita3k/vita3k/commit/d2e99945
- CPU backend: Dynamic  

# Test environment summary
- Tested by: othmanead
- OS: Windows 11
- CPU: Intel i5-10300H
- GPU: RTX 3060 mobile
- RAM: 16 GB

# Issue: 
- Game reaches menu but crashes after

# Screenshots:
![image](https://user-images.githubusercontent.com/106798110/172984586-329e5b71-057b-4561-9ed3-925877ba075a.png)

# Log

[vita3k.log](https://github.com/Vita3K/compatibility/files/8875516/vita3k.log)

# Recommended labels
- Menu
- Crash

**Comment by ArbestRiAgain** [comment]
Goes further after latest update, but appears to spam errors a lot.
![image](https://github.com/Vita3K/compatibility/assets/132712354/3d1a279b-ea0c-4ee7-85a9-2107a4433d56)
[Log file](https://github.com/Vita3K/compatibility/files/11640087/vita3k.zip)

**Comment by YousefSlimani99** [comment]
Can you please fix it? I was playing this game recently and I have the same error

**Comment by ShadowFox02** [comment]
Upon updating to version 0.2.0 3563-b5aa1ec5, the game suddenly went past the title screen and made it to the main menu. However, it crashed again after trying to start Voyage mode. I tried booting it up again a few times, but it just went back to crashing at the title screen.
[Log file](https://github.com/Vita3K/compatibility/files/14134695/PCSE00009.-.Lumines.Electronic.Symphony.-.Copy.log)

**Comment by zizhengwu** [comment]
looks like I can start voyage mode:

![image](https://github.com/Vita3K/compatibility/assets/2246655/757cb395-83e6-47b8-9961-ad5b38b6bde2)

**Comment by YousefSlimani99** [comment]
> looks like I can start voyage mode:
> 
> ![image](https://private-user-images.githubusercontent.com/2246655/302119990-757cb395-83e6-47b8-9961-ad5b38b6bde2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2OTA4MTksIm5iZiI6MTcxMTY5MDUxOSwicGF0aCI6Ii8yMjQ2NjU1LzMwMjExOTk5MC03NTdjYjM5NS04M2U2LTQ3YjgtOTk2MS1hZDViMzhiNmJkZTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjlUMDUzNTE5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDUwZmE1NzBlNTgyOWZhOTZlZjNjNDViZWUwN2ZlZDZmYWE3NDE2NTRlYzUwMTlmNWU2NTY3NDEwYjExNDIzMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.EfW0M3QtKitEI9CMJuKX1fpSNDa5JZ9FZMdkzM0yP5k)
> 
> edit: looks like only first boot works. not sure if save data is the reason, following boots crashed at title screen



> looks like I can start voyage mode:
> 
> ![image](https://private-user-images.githubusercontent.com/2246655/302119990-757cb395-83e6-47b8-9961-ad5b38b6bde2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2OTA4NDcsIm5iZiI6MTcxMTY5MDU0NywicGF0aCI6Ii8yMjQ2NjU1LzMwMjExOTk5MC03NTdjYjM5NS04M2U2LTQ3YjgtOTk2MS1hZDViMzhiNmJkZTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjlUMDUzNTQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OThlMDdjZDZhZWE2ODU5ZDlmNWFhMjYxYTRiZGQ3ODkxNTM1Y2ZmZjhiYzZmOWVhNTllZTU5MzJjYTg1MmUxYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.mVnWPvmUuVyToGf4ysbIV597kRzI1vNQI_PbeV8PBtQ)
> 
> edit: looks like only first boot works. not sure if save data is the reason, following boots crashed at title screen

How did you managed to do that?!

**Comment by YousefSlimani99** [comment]
> looks like I can start voyage mode:
> 
> ![image](https://private-user-images.githubusercontent.com/2246655/302119990-757cb395-83e6-47b8-9961-ad5b38b6bde2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2OTA4NDcsIm5iZiI6MTcxMTY5MDU0NywicGF0aCI6Ii8yMjQ2NjU1LzMwMjExOTk5MC03NTdjYjM5NS04M2U2LTQ3YjgtOTk2MS1hZDViMzhiNmJkZTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjlUMDUzNTQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OThlMDdjZDZhZWE2ODU5ZDlmNWFhMjYxYTRiZGQ3ODkxNTM1Y2ZmZjhiYzZmOWVhNTllZTU5MzJjYTg1MmUxYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.mVnWPvmUuVyToGf4ysbIV597kRzI1vNQI_PbeV8PBtQ)
> 
> edit: looks like only first boot works. not sure if save data is the reason, following boots crashed at title screen

How did you managed to do that?!

**Comment by ShadowFox02** [comment]
As of https://github.com/Vita3K/Vita3K/commit/dbe6951371bd573f11962464d393a2eb552fa48f, the game once again goes in-game on the first launch after updating. This time, I was actually able to reach gameplay.

- The main menu and all foreground elements appeared to display perfectly
- The first level, "The Future of the Future", did not display any background effects at all, but appeared to look fine otherwise.
- The second level, "Good Girl", had noticeable flickering issues with it's background.
- Third level, "Moistly", appeared to display perfectly.
- The game crashed loading into the fourth level, "4 AM".
- Upon relaunching the game, it just freezes at the same place it has been

Here are my specs:
- Version: v0.2.0
- Build number: 3615
- Commit hash: https://github.com/Vita3K/Vita3K/commit/53741632bc98ebd3770cbe55b723843e1b83a458
- CPU backend: Dynarmic.
- GPU backend: Vulkan
### Test Environment Summary:
- OS: Windows 10 22H2
- CPU: Intel i9-11900K
- GPU: NVIDIA GeForce RTX 3070
- RAM: 32GB

![image2](https://github.com/Vita3K/compatibility/assets/25962188/00086b24-dc75-46a8-b480-34f619de2238)
![image1](https://github.com/Vita3K/compatibility/assets/25962188/a17c8e72-5720-4e0d-b7b6-7c50d76b2952)

**Comment by boomercore** [comment]
Unsure of how you guys are getting this far into it.  Latest build gives me the endless loading circle as seen in 
https://github.com/Vita3K/compatibility/issues/693#issuecomment-1574499069
and error spam.  Tried game versions 1.00 and 1.01.

![image](https://github.com/user-attachments/assets/dace919b-e62d-4027-ad68-bb3b157ea119)

**Comment by YousefSlimani99** [comment]
For crying out loud, why this game wasn't fixed yet?!

**Comment by zizhengwu** [comment]
> Unsure of how you guys are getting this far into it

From my experience, I could only "launch Lumines Electronic Symphony and start the voyage mode" on a fresh installation of Vita3K & first-time boot of Lumines Electronic Symphony.

After that, all following boots crashed. I'm not sure why. Maybe people can try and share theories.

**Comment by ShadowFox02** [comment]
Was randomly able to once again get in-game on the first boot on today's update, https://github.com/Vita3K/Vita3K/commit/1c315b1f6d66ce03b729f910df6241c59ad0e31c.
I managed to get footage of being in-game [here](https://www.youtube.com/watch?v=hBQppXhYC-s).
Log file: [PCSE00009 - [Lumines Electronic Symphony].log](https://github.com/user-attachments/files/16703401/PCSE00009.-.Lumines.Electronic.Symphony.log)

**Comment by ShadowFox02** [comment]
Got in-game once again on https://github.com/Vita3K/Vita3K/commit/9143a14fcd2c1cf7732f92bde87432916b169e9d

**Some Observations:**
-Saving and Loading appear to work just fine.
-Some update between my last report and now has seemingly fixed the backgrounds in tFotF, Moistly, and Master 1, which now appear to render perfectly.
-Stopwatch and Single Skin modes appear to work perfectly.
-Chain blocks in Master Mode _might_ be fixed, not completely sure on that though.
-Master 2 still causes a crash. Subsequent boot ups still lock up unfortunately.

Wasn't able to save the log this time, sorry.

**Comment by mischreiber** [comment]
Same story here. I got into the game once long ago, but never since. Current build (3690 https://github.com/Vita3K/Vita3K/commit/f02851437d8bce2ebf54cd6d4922cd4b0faba654) still hangs indefinitely after the menu.

**Comment by KazumiA1** [comment]
Using build (3690 Vita3K/Vita3k@ f0285143)

I got curious and started deleting stuff, like the license for Lumines Electronic Symphony and I was able to get past the menu. Only downside is that it was only the demo and crashed on the 3rd level of the trial version.
https://www.youtube.com/watch?v=-lafd3jeX4o
NOTE : this is only the demo version **NOT THE FULL GAME**

**Comment by ShadowFox02** [comment]
Using build https://github.com/Vita3K/Vita3K/commit/85c9f978ad978f51138fdabc19e6dc42b26d1fcf.

Can confirm that switching to trial mode makes the game consistently go in-game. Additionally, switching the CPU Backend to _Unicorn_ fixes the crash on the third stage, effectively making the demo fully playable. It does introduce slowdown on the first stage though.

**Comment by Flannels** [comment]
Has some way to get the game working again been found? I can't figure it out

**Comment by skullthug** [comment]
Looks like as of 0.2.0.3807 this is still broken, I just get the endless spinner past the title screen and the console spamming errors

**Comment by KJF-dev** [comment]
I was messing around with my Vita3k settings trying to get it to run and to my surprise, it seems to be fully working.

I enabled "Disable Surface Sync" and "FPS Hack" in GPU settings and boot the game. After this, it loads into the main menu and allows you to start the voyage.

<img width="1860" height="1086" alt="Image" src="https://github.com/user-attachments/assets/2d9c8c6f-c2c9-4299-9314-9faa36d0936d" />

Opening World Block softlocks the game but enabling "Signed in PSN" fixes it (it continuously flashes black with it enabled in the Main Menu though, epilepsy warning).

Changing the settings back to the original after this also seems to make the fix persist. I get some minor graphical glitches when entering menus but everything else seems to be fine.

This works both on my Steam Deck and PC (both Vita3k version v0.2.0 3807-4e526f0c).
Will report back if I experience any other problems.

**Comment by zylos146** [comment]
When using a 100% save file and the playlist mode, none of the songs trigger a crash. I double checked the 4th level song, 4AM, and it didn't crash. It appears something in the game logic is triggered on the 3 -> 4 transition which causes the failure.

This means you can likely use the Playlist mode without issue. If you have a 100% save, you could even just re-create the original song order and play that instead of the main Voyage mode.

This also hopefully means the crash is just from some game logic, rather than anything relating to the graphics which would be harder for me to diagnose.

This may be the origin, it's the last non-`gl::draw` log entry before the crash. I don't have what code address caused this failure though. It doesn't appear to output it unlike the `ArmDynarmicCallback::MemoryWrite` failure
```
|C| [logging::exception_handler]: Exception EXCEPTION_ACCESS_VIOLATION (0xC0000005).
|C| [logging::exception_handler]: Read violation at address 0x8.
```

**Comment by ShadowFox02** [comment]
Using your patch, the skin crash does not occur if using the old Unicorn backend. I was able to get ~2/3 of the way through Voyage before the game randomly crashed on _Wooden Toy_. Some things to note on Unicorn:
- Loading screens take noticably longer
- Certain skins (ones with a lot of 3D objects?) have heavily degraded performance.
- Sometimes, skin backgrounds will glitch out and only appear as a few static lines on the top.

Other observations when switching back to Dynarmic:
- Chain blocks still erroneously still appear in Master Mode. Also occurs in Unicorn.
- Skin crash can still occur in Playlist mode, as well as on Master mode. Not sure on specifics though.
- Menus being touch-screen only is an intentional part of the game.

**Comment by zylos146** [comment]
It being a skin related crash makes sense. Do you have a log file from getting to Wooden Toy on Unicorn? I'm really interested in seeing if a Kernal Alloc error appears on Unicorn. I was able to isolate a normal level transition in the logs

```
[20:28:17.008] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[20:28:17.029] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[20:28:17.043] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[20:28:17.049] |E| [stat_file]: Missing file at "C:/Users/XXXX/AppData/Roaming/Vita3K/Vita3K/ux0/app/PCSE00009/data/skin/skin101/SkinSettings_06.setting" (target path: app0:data/skin/skin101/SkinSettings_06.setting)
[20:28:17.049] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[20:28:17.051] |E| [stat_file]: Missing file at "C:/Users/XXXX/AppData/Roaming/Vita3K/Vita3K/ux0/app/PCSE00009/data/skin/skin101/bg/bgsphere.curve" (target path: app0:data/skin/skin101/bg/bgsphere.curve)
[20:28:17.051] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[20:28:17.057] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[20:28:17.060] |W| [MjpegDecoderState::send]: Error sending Mjpeg packet: Invalid Data (AVERROR_INVALIDDATA).
[20:28:17.060] |W| [MjpegDecoderState::receive]: Error receiving Mjpeg frame: Requires Another Call (AVERROR(EAGAIN)).
[20:28:17.061] |W| [MjpegDecoderState::send]: Error sending Mjpeg packet: Invalid Data (AVERROR_INVALIDDATA).
[20:28:17.061] |W| [MjpegDecoderState::receive]: Error receiving Mjpeg frame: Requires Another Call (AVERROR(EAGAIN)).
[20:28:17.062] |E| [export_sceKernelAllocMemBlock::<lambda_1>::operator ()]: sceKernelAllocMemBlock returned SCE_KERNEL_ERROR_INVALID_ARGUMENT (0x80020005)
[20:28:17.063] |E| [export_sceGxmMapMemory::<lambda_1>::operator ()]: sceGxmMapMemory returned SCE_GXM_ERROR_INVALID_POINTER (0x805B0004)
[20:28:17.078] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[20:28:17.078] |W| [MjpegDecoderState::send]: Error sending Mjpeg packet: Invalid Data (AVERROR_INVALIDDATA).
[20:28:17.078] |W| [MjpegDecoderState::receive]: Error receiving Mjpeg frame: Requires Another Call (AVERROR(EAGAIN)).
[20:28:17.090] |E| [stat_file]: Missing file at "C:/Users/XXXX/AppData/Roaming/Vita3K/Vita3K/ux0/app/PCSE00009/data/blk_eff/blk_S/blk_S.link" (target path: app0:data/blk_eff/blk_S/blk_S.link)
[20:28:17.090] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[20:28:17.092] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[20:28:17.092] |W| [MjpegDecoderState::send]: Error sending Mjpeg packet: Invalid Data (AVERROR_INVALIDDATA).
[20:28:17.092] |W| [MjpegDecoderState::receive]: Error receiving Mjpeg frame: Requires Another Call (AVERROR(EAGAIN)).
[20:28:17.093] |E| [stat_file]: Missing file at "C:/Users/XXXX/AppData/Roaming/Vita3K/Vita3K/ux0/app/PCSE00009/data/skin/skin101/block/box_red.link" (target path: app0:data/skin/skin101/block/box_red.link)
[20:28:17.093] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[20:28:17.094] |E| [stat_file]: Missing file at "C:/Users/XXXX/AppData/Roaming/Vita3K/Vita3K/ux0/app/PCSE00009/data/skin/skin101/block/box_whi.link" (target path: app0:data/skin/skin101/block/box_whi.link)
[20:28:17.094] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[20:28:17.094] |E| [stat_file]: Missing file at "C:/Users/XXXX/AppData/Roaming/Vita3K/Vita3K/ux0/app/PCSE00009/data/skin/skin101/block/blk_del.link" (target path: app0:data/skin/skin101/block/blk_del.link)
[20:28:17.094] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[20:28:17.095] |E| [stat_file]: Missing file at "C:/Users/XXXX/AppData/Roaming/Vita3K/Vita3K/ux0/app/PCSE00009/data/light/light_def.link" (target path: app0:data/light/light_def.link)
[20:28:17.095] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[20:28:17.096] |E| [stat_file]: Missing file at "C:/Users/XXXX/AppData/Roaming/Vita3K/Vita3K/ux0/app/PCSE00009/data/basic_grids/gray.link" (target path: app0:data/basic_grids/gray.link)
[20:28:17.096] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[20:28:17.106] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[20:28:17.126] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[20:28:17.140] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[20:28:17.161] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[20:28:17.175] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
```

Comparing to the crashed log, it looks like it crashes right at the point where this normal transition emits
```
[20:28:17.062] |E| [export_sceKernelAllocMemBlock::<lambda_1>::operator ()]: sceKernelAllocMemBlock returned SCE_KERNEL_ERROR_INVALID_ARGUMENT (0x80020005)
[20:28:17.063] |E| [export_sceGxmMapMemory::<lambda_1>::operator ()]: sceGxmMapMemory returned SCE_GXM_ERROR_INVALID_POINTER (0x805B0004)
```

This makes me think it's some allocator error, which could potentially mean it fails after switching a certain amount of levels out for new ones. It may fail an allowable amount for a few levels, but then fails after that.

That would also make sense on why my initial Playlist testing worked. I dragged each song into a single song playlist, played, then quit, and dragged it out to replace with the next song. That wouldn't catch anything based on having to load multiple songs. I was only intending to test if it was an asset / graphics / shader issue or not.

It's been like nearly a decade since I've played this, so I had forgotten the menu was touch screen only.

**Comment by zylos146** [comment]
This appears to be the equivalent Unicorn log from the **Run 1** log file you linked

```
[19:09:35.115] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:09:35.226] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:09:35.268] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/skin/skin124/SkinSettings_06.setting" (target path: app0:data/skin/skin124/SkinSettings_06.setting)
[19:09:35.268] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[19:09:35.323] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:09:35.347] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/skin/skin124/bg/bgsphere.curve" (target path: app0:data/skin/skin124/bg/bgsphere.curve)
[19:09:35.348] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[19:09:35.420] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:09:35.531] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:09:35.567] |W| [MjpegDecoderState::send]: Error sending Mjpeg packet: Invalid Data (AVERROR_INVALIDDATA).
[19:09:35.567] |W| [MjpegDecoderState::receive]: Error receiving Mjpeg frame: Requires Another Call (AVERROR(EAGAIN)).
[19:09:35.622] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:09:35.729] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/blk_eff/blk_S/blk_S.link" (target path: app0:data/blk_eff/blk_S/blk_S.link)
[19:09:35.730] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[19:09:35.733] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:09:35.761] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/skin/skin124/block/box_red.link" (target path: app0:data/skin/skin124/block/box_red.link)
[19:09:35.762] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[19:09:35.771] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/skin/skin124/block/box_whi.link" (target path: app0:data/skin/skin124/block/box_whi.link)
[19:09:35.771] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[19:09:35.777] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/skin/skin124/block/blk_del.link" (target path: app0:data/skin/skin124/block/blk_del.link)
[19:09:35.777] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[19:09:35.783] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/light/light_def.link" (target path: app0:data/light/light_def.link)
[19:09:35.783] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[19:09:35.791] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/basic_grids/gray.link" (target path: app0:data/basic_grids/gray.link)
[19:09:35.792] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[19:09:35.823] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:09:35.920] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
```

This shows some Mjpeg decoder errors, but not as many. The kernal allocation errors are almost completely gone, but does manifest once, before the snippet I posted above

```
[18:49:29.004] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[18:49:29.101] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[18:49:29.119] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/skin/skin003/SkinSettings_04.setting" (target path: app0:data/skin/skin003/SkinSettings_04.setting)
[18:49:29.119] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[18:49:29.185] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[18:49:29.193] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/skin/skin003/bg/bgsphere.curve" (target path: app0:data/skin/skin003/bg/bgsphere.curve)
[18:49:29.193] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[18:49:29.202] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/skin/skin003/bg/style.curve" (target path: app0:data/skin/skin003/bg/style.curve)
[18:49:29.202] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[18:49:29.275] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[18:49:29.372] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[18:49:29.438] |W| [MjpegDecoderState::send]: Error sending Mjpeg packet: Invalid Data (AVERROR_INVALIDDATA).
[18:49:29.438] |W| [MjpegDecoderState::receive]: Error receiving Mjpeg frame: Requires Another Call (AVERROR(EAGAIN)).
[18:49:29.438] |E| [export_sceKernelAllocMemBlock::<lambda_1>::operator ()]: sceKernelAllocMemBlock returned SCE_KERNEL_ERROR_INVALID_ARGUMENT (0x80020005)
[18:49:29.438] |E| [export_sceGxmMapMemory::<lambda_1>::operator ()]: sceGxmMapMemory returned SCE_GXM_ERROR_INVALID_POINTER (0x805B0004)
[18:49:29.456] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[18:49:29.495] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/blk_eff/blk_S/blk_S.link" (target path: app0:data/blk_eff/blk_S/blk_S.link)
[18:49:29.495] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[18:49:29.535] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/skin/skin003/block/box_red.link" (target path: app0:data/skin/skin003/block/box_red.link)
[18:49:29.536] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[18:49:29.543] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/skin/skin003/block/box_whi.link" (target path: app0:data/skin/skin003/block/box_whi.link)
[18:49:29.544] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[18:49:29.546] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[18:49:29.553] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/skin/skin003/block/blk_del.link" (target path: app0:data/skin/skin003/block/blk_del.link)
[18:49:29.553] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[18:49:29.559] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/light/light_def.link" (target path: app0:data/light/light_def.link)
[18:49:29.560] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[18:49:29.569] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/basic_grids/gray.link" (target path: app0:data/basic_grids/gray.link)
[18:49:29.569] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[18:49:29.629] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[18:49:29.720] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[18:49:29.838] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
```

SkinSettings_06 appears to not result in any kernel allocation errors, but SkinSettings_04 does result in one. The cause of the crash for Run 1 is

```
[19:16:20.182] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:16:20.217] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:16:20.249] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/skin/skin117/SkinSettings_04.setting" (target path: app0:data/skin/skin117/SkinSettings_04.setting)
[19:16:20.249] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[19:16:20.251] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:16:20.286] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:16:20.300] |E| [stat_file]: Missing file at "D:/Vita3K Directory/Vita3K/ux0/app/PCSE00009/data/skin/skin117/bg/bgsphere.curve" (target path: app0:data/skin/skin117/bg/bgsphere.curve)
[19:16:20.300] |W| [io_error_impl]: stat_file (_sceIoGetstat) returned 0x80010002
[19:16:20.321] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:16:20.363] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:16:20.394] |W| [MjpegDecoderState::send]: Error sending Mjpeg packet: 0xFFFFFFD8.
[19:16:20.394] |W| [MjpegDecoderState::receive]: Error receiving Mjpeg frame: Requires Another Call (AVERROR(EAGAIN)).
[19:16:20.404] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:16:20.439] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:16:20.481] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:16:20.497] |C| [logging::exception_handler]: Exception EXCEPTION_ACCESS_VIOLATION (0xC0000005). 
[19:16:20.497] |C| [logging::exception_handler]: Read violation at address 0x10.
[19:16:20.529] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:16:20.571] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
[19:16:20.620] |W| [renderer::gl::sync_clipping]: Unimplemented region clip mode used: SCE_GXM_REGION_CLIP_INSIDE
```

Which matches the above theory about SkinSettings_04 being the issue for Unicorn. It's interesting that Unicorn made the errors for SkinSettings_06 disappear, but not 04. I had already started looking at the Skin / Asset loader, so I have a good starting point, but its a lot more complex than the Content Rating / TSS checks.

It's interesting that these failures don't appear to be guaranteed on Unicorn, that you reached Wooden Toy on one run, but Embracing the Future on another run. **Run 2** failed with the same SkinSettings_04 issue as **Run 1**

**Comment by zylos146** [comment]
It doesn't appear to crash due to a kernal allocation error, but rather a JPEG decoding issue when loading song backgrounds. 
It's throwing the exception within FFMPEG/libavcodec's get_vlc2 which handles the Variable-Length Coding for the JPEG.

The console throws a lot of bad VLC errors, but apparently one of them is bad enough to trigger some read violation probably through a bad pointer. 

<img width="1088" height="257" alt="Image" src="https://github.com/user-attachments/assets/9f362f28-a695-4290-9304-73a5f6c506b8" />

<img width="477" height="82" alt="Image" src="https://github.com/user-attachments/assets/76f6de75-1d3a-4b1d-90a0-9ca52bc839b0" />

The specific JPEG that was in the buffer at the time of loading was 

![Image](https://github.com/user-attachments/assets/9e80ba1c-e33f-4ca6-9211-211bf758a453)

This decodes fine in the windows image viewer.
Running the JPEG through ffmpeg on my desktop and asking it to re-compress the image apparently doesn't crash it.
I'm curious if this is some issue with how the data is being sent to libavcodec, but it will take more poking around.

I still don't know why the crashing level was inconsistent on Unicorn. I would have expected a decoding error like this to be a consistent issue when the asset in question loaded. There's also the question of why each individual song + assets can be loaded when played individually through Playlist Mode

**Comment by ShadowFox02** [comment]
Using the new patch, I was able to do a full run of Master Mode with no crashes. Stopwatch mode does not have skin transitions and has previously worked perfectly even before the patches. Haven't tested Voyage or Playlist yet, but should be able to later, will update this post with results for those.

**Comment by zylos146** [comment]
First PR is up. This completely resolved the background loading issue. The second patch of mine is no longer needed. The only patch required is the Content Rating one still. That patch is available here, [PCSE00009.txt](https://github.com/user-attachments/files/22013673/PCSE00009.1.txt)

The Content Rating issue will take some time to resolve, as it's related to un-implemented APIs.
With the patch above though, this game is fully playable as far as I can tell.

**Comment by YousefSlimani99** [comment]
> First PR is up. This completely resolved the background loading issue. The second patch of mine is no longer needed. The only patch required is the Content Rating one still. That patch is available here, [PCSE00009.txt](https://github.com/user-attachments/files/22013673/PCSE00009.1.txt)
> 
> The Content Rating issue will take some time to resolve, as it's related to un-implemented APIs. With the patch above though, this game is fully playable as far as I can tell.

And where I can put this txt file anyway?!

**Comment by zylos146** [comment]
The PR has been merged in, so you can just run the `update-vita3k` script in your Vita3k folder. This will update to the latest master build. No need for the custom builds I linked above, although they will still work.

I'm poking around potential solutions for the ContentRating issue. It appears it would require providing some sort of implementation of the `sceNpTus` and `sceNpTss` libraries which are currently entirely un-implemented. They are complex network storage relating to online scoreboard and other online functionality.

I'll continue poking around for a quick solution, like figuring out what the correct error code responses would be for the Title Storage APIs if the user was not connected to the network. If I can fake lacking network connectivity, that would be an easy solution that shouldn't disrupt other games either.

**Comment by ShadowFox02** [comment]
Apologies for the long response wait.
Completed a full run of Voyage. I did have a some minor slowdown on the very first skin (which was noticeably absent when returning to it after completing a loop), but other than that, it ran full-speed. A few minor issues I have noticed here and playing previously:
- Graphics corruption issues on the Main Menu. Some of the corruptions resembled 3D set-pieces from certain skins, e.g. the giant flower from _Aganju_. Not consistent, did not occur when playing today.
- Odd, white flickering in the backgrounds of certain skins. This has been observed in in the skins _Good Girl_, _Hey Boy, Hey Girl_, and _Out of the Blue_. Does not seem to occur if the skin is the first one loaded after starting a new game.
- When attempting to start a game, there seems to be very small chance for the emulator to crash on a Read Access Error.

Other than that and the aforementioned _Chain Blocks in Master Mode_ bug, the game works perfectly.

I'm running on:
- Windows 10
- Intel i9-11900K
- NVIDIA RTX 3070
- 32GB RAM

**Comment by skullthug** [comment]
Hmm I've run the update-vita3k.bat and verified I'm on the latest version, however I'm still unable to get past the title screen. I still just get the endless spinner ghost & boomercore posted about, while the error

`[15:16:42.935] |C| [ArmDynarmicCallback::MemoryRead]: PC is 0x0
[15:16:42.935] |E| [ArmDynarmicCallback::MemoryRead]: Invalid read of uint32_t at address: 0x1c50
PC: 0x00000000,   SP: 0x8014bf18,   LR: 0x80011c21
r0 : 0x000153ac   r6 : 0x00000000
r1 : 0x00000000   r7 : 0x00000000
r2 : 0x00000000   r8 : 0x00000000
r3 : 0x00000039   r9 : 0x886399e0
r4 : 0x00000000   r10: 0x00000000
r5 : 0x00000000   r11: 0x00000000
r12: 0x80011cd1
Thumb: false`
spams in the console window

Not sure if I missed something?

**Comment by nadia1461** [comment]
@skullthug 

You need to be on the PR version linked above. It has not been merged into master as I am working on the correct approach to ensure regressions don't occur other games.

> The last PR is up. It should be mentioned right before this comment. Once this is merged in, the game will be `Playable` rather than `InGame+`.
> 
> All non-online functionality, aside from minor bugs as reported by ShadowFox02, will be available. The online functionality is secondary from the primary game play and would also require a significant investment to properly implement the Vita TSS/TUS servers.
> 
> The PR Build can be used until this is merged into master. **No Patches are required anymore if using the PR Build**
> 
 [Vita3k Windows: TSS/TUS Graceful Unimplementation](https://github.com/Vita3K/Vita3K/actions/runs/17443530869/artifacts/3920260429)

 [Vita3k Linux: TSS/TUS Graceful Unimplementation](https://github.com/Vita3K/Vita3K/actions/runs/17443530869/artifacts/3920232369)

**Comment by nadia1461** [comment]
@KazumiA1 
Yes, the No Background patch will remove the background clips. Try using the PR build linked above and removing all the patches. It should work with the backgrounds loading

**Comment by nadia1461** [comment]
I've tested the Linux build on the Steam Deck and I can confirm it both fully works (with the above build) and appears to have no major performance problems (outside the stage before Wooden Toy having significant slowdowns, even on PC/Windows)

**Comment by Airwalk540** [comment]
Tested the Linux build on Steam Deck with the Japanese copy of Lumines and can confirm it works with that version as well. Amazing. Some slowdown here and there, has trouble quitting a run early, etc but overall it works.

Any indication when this fix will be rolled into the main version?

**Comment by Takanu** [comment]
It's incredible to be able to play this game again, thank you!

**Comment by nadia1461** [comment]
https://github.com/Vita3K/Vita3K/pull/3657 has been merged

This means Lumines should now qualify as `Playable`, rather than `Ingame +`
There are several user reports above which can verify this new status. 
I have also confirmed as well that with this new release Lumines is fully playable

No patches are required to play using the new Vita3k V0.2.0 3830
[Vita3k Windows: Release 3830](https://github.com/Vita3K/Vita3K/actions/runs/19275984356/artifacts/4535573600)

You can also just update to 3830 via the update-vita3k script bundled with every vita3k install, or via the update dialog which appears when you launch Vita3k.

**Comment by develm** [comment]
Is there a way to do this on the Android version of the emulator?

**Comment by csrednicki** [comment]
Getting some weird video glitches when entering Options. There is also strobe light effect of background starting level 2 (Benny Benassi - Good girl).

<img width="977" height="585" alt="Image" src="https://github.com/user-attachments/assets/f052e300-9f5b-45d6-a717-c2a26e63e207" />

<img width="972" height="580" alt="Image" src="https://github.com/user-attachments/assets/b3d641f0-70a1-4320-820d-968ddfd44e72" />

<img width="974" height="590" alt="Image" src="https://github.com/user-attachments/assets/398dabe6-03bf-47dd-b506-919eda5b4690" />

# Vita3K summary
- OS: MacOS Tahoe 26.2
- Version: v0.2.0
- Build number: 3845
- Commit hash: https://github.com/vita3k/vita3k/commit/f293954a
- GPU backend: Vulkan

**Comment by Gerold55** [comment]
So this does work but has some flickering when in the menus

**Comment by csrednicki** [comment]
Was working fine with version 3943, no flickering, no other issues. Starting from 3944 there is audio stutter right from the start of the game. Tested on POCO F8 Ultra (Ardeno 840).

**Comment by nadia1461** [comment]
[csrednicki](https://github.com/csrednicki), The audio stuttering appears to be a known android audio issue which is currently being worked on by the vita3k devs. It's affecting other games as well, and is in active discussion in their discord. I can look at the other issues, but the audio stuttering is likely something you'll need to wait for a general android fix for.

> Was working fine with version 3943, no flickering, no other issues. Starting from 3944 there is audio stutter right from the start of the game. Tested on POCO F8 Ultra (Ardeno 840).

Granted, the reported audio issues started with 3850, so not 3943, but it is in line with them knowing android audio has issues and is an active area of work. So I may not be 100% correct on the current state, but it does sounds like an area where I might not be of as much help

**Comment by nadia1461** [comment]
[Gerold55](https://github.com/Gerold55)
> So this does work but has some flickering when in the menus

Would you be able to show a short clip or screenshots of what the flickering looks like?
That would help greatly with knowing when I've got replication on my side so I can look into the issue.
What are your hardware details, are you using PC or android?

Does the flickering look like the previously reported flickering above on the main menu where white blocks appear?

**Comment by nadia1461** [comment]
[RobertoTorino](https://github.com/RobertoTorino)
> I played this game for an hour yesterday, but with the latest version i'm getting the endless loop again, it will not even start. Installed VITA3K from scratch, tried different verions of the game (pkg zip) but no success. What has changed since 3830?

Does 3830 still work for playing the game? Would you be able to upload the log file for vita3k (located in the vita3k folder housing the executable) or a snippet of the last section of the log file if it's too large. 

Is this the same endless loop where you get stuck right before it loads into the main menu?
Does it work on versions after 3830 if you apply the patch files I listed above?

One of those patches completely bypasses the TUS checks during the loading screen to main menu, which should allow it to be used even on unfixed versions of vita3k

**Comment by nadia1461** [comment]
[ZaptorZap](https://github.com/ZaptorZap), I'm not exactly sure where the artifacts are in the screenshots you posted?
The background is quite noisy, so while I trust it's flickering, it's hard for me to notice what is wrong.
I may also just be missing something obvious. If you could help describe what the flickering and artifacting looks like more, it would greatly help. Even just drawing over the screenshot to show the flickering area would help. 

I can attempt replication on linux as well to see if I notice anything.
The main menu transition flickering sounds similar to that reported by [csrednicki](https://github.com/csrednicki), which is something I can look into.

> Here to repeat the reports on Linux with the latest version(`v0.2.1 3916-1167cb63`) regardless of renderer(GL or Vulkan). I get flickering, yet only on _Moistly_ (3rd stage in voyage) as well as main menu transitions.
> 
> ```
> |W| [sync_clipping]: STUB SCE_GXM_REGION_CLIP_INSIDE
> Unhandled SIGSEGV at rip 0x000055cdaefc9cce
> |W| [sync_clipping]: STUB SCE_GXM_REGION_CLIP_INSIDE
> Unhandled SIGSEGV at rip 0x000055cdaefc9cce
> |W| [sync_clipping]: STUB SCE_GXM_REGION_CLIP_INSIDE
> Unhandled SIGSEGV at rip 0x000055cdaefc9cce
> ```
> 
> Is generally present at all times in the terminal, and during this stage the below artifacts render about every other frame.

**Comment by csrednicki** [comment]
@nadia1461 I have some graphical issues for fonts on macos version 3943 (last one with not bugged sound stutter).

<img width="953" height="565" alt="Image" src="https://github.com/user-attachments/assets/67fd4164-d344-4cae-9bed-6f2638b8d51e" />

**Comment by csrednicki** [comment]
version 3956, sound is good, no stutter, only fonts issue

<img width="962" height="573" alt="Image" src="https://github.com/user-attachments/assets/18447d35-539c-4879-b9b2-e90d0e9d1424" />

**Comment by ZaptorZap** [comment]
> [ZaptorZap](https://github.com/ZaptorZap), I'm not exactly sure where the artifacts are in the screenshots you posted? The background is quite noisy, so while I trust it's flickering, it's hard for me to notice what is wrong. I may also just be missing something obvious. If you could help describe what the flickering and artifacting looks like more, it would greatly help. Even just drawing over the screenshot to show the flickering area would help.

@nadia1461 • Apologies for the inactivity. I had some difficulty in re-dumping the game for some reason, but we're alright now. Here's a recording of the stage using v0.2.1-4056-5b040dc4 ↓ **Flashing lights warning**

https://github.com/user-attachments/assets/4a27bdb8-c737-4231-9d5c-c647c5d3ae9f

The problem has noticeably been improved since last I tried, but still present. I would not say the Chain Block powerup is what causes the flashing lights to occur, though it is an interesting coincidence in this recording. The SEGVs occur in this pattern now: `0x00007fefdf76ecf7 → 0x00007fefdf76ecfb → 0x00007fefdf76ed05 → 0x00007fefdf76ecfb → 0x00007fefdf76ecf7 → 0x00007fefdf76ecfb → 0x00007fefdf76ed05 …` (game version 1.0.1 • `eboot.bin` MD5 hash: `0a28e048cd091672fa6a847fdca7bb7b`)
